src/cpp/include/openvino/genai/text2image/clip_text_model_with_projection.hpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <string>
+
+#include "openvino/genai/visibility.hpp"
+#include "openvino/genai/tokenizer.hpp"
+
+#include "openvino/core/any.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "openvino/runtime/infer_request.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace genai {
+
+class OPENVINO_GENAI_EXPORTS CLIPTextModelWithProjection {
```

Comment:
in a separate PR let's add LoRA support to this text encoder similarly to https://github.com/openvinotoolkit/openvino.genai/pull/911/files#diff-8f2fbf9a7bec54e768bb14ad1450b90f2f66953a0e66f57d364ce72cc4e67cf2